hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSConfigKeys.java

```diff
@@ -1557,6 +1557,17 @@
   public static final double
       DFS_DATANODE_RESERVE_FOR_ARCHIVE_DEFAULT_PERCENTAGE_DEFAULT = 0.0;
 
+
+  public static final String
+      DFS_RESOLVE_NAMESERVICE_NEEDED =
+      "dfs.resolve.nameservice.needed";
```

Comment:
"dfs.nameservices.resolution.enabled" or "dfs.nameservices.resolution-enabled" may be a better property name